### DISCUSSION TITLE
Error returned when using OpenFolderPickerAsync.

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#12797] [Q&A]
In Avalonia, when I use await StorageProvider.OpenFolderPickerAsync(new FolderPickerOpenOptions), it returns a result normally when selecting folders in the dialog. However, the issue arises when I start selecting folders with spaces in their names. When I use `Console.WriteLine()` to print the path that the system has collected when I select a folder with spaces in the dialog, I notice that the spaces in the folder names have been replaced with "%20". I'm not sure what's causing this error. I had to use Contains to detect which folder paths contain spaces and then replace "%20" with actual spaces to correct it.

1. Go to StorageProvider.OpenFolderPickerAsync(new FolderPickerOpenOptions);
2. Click on Select a folder contains spaces.
3. See error 'Could not find a part of the path'

**Screenshots**
![image](https://github.com/AvaloniaUI/Avalonia/assets/85250803/86d65de3-b228-4142-9dd6-c0e529630645)

- OS: [Windows 11]
- Version [22H2]

**Comment by timunie** [comment]
please file a minimal sample (at least the snippet how your select the folders) to reproduce as it works on my end using Window 11. Also let us know your Avalonia version

**Comment by ghost** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/85250803/af8b60c6-d6cd-434d-bf8e-025b486a41f5)
![image](https://github.com/AvaloniaUI/Avalonia/assets/85250803/73709ad1-2d90-41a0-8329-7dfa9e22c551)

I'm using Avalonia 11.0.2

**Comment by stevemonaco** [comment]
I use `LocalPath` here instead of `AbsolutePath` to avoid the escaped spaces (%20). It still provides the full path from the drive letter. Desktop app, so I'm not sure how that fares if you're doing mobile.

**Comment by timunie** [comment]
okay so you are using `AbsolutePath` of the URI which is escaped by default iirc. `LocalPath` is unescaped and should have the whitespaces print as expected. This is by design of .NET and out of Avalonias responsibilty: https://learn.microsoft.com/en-us/dotnet/api/system.uri?view=net-7.0 

serching the internet results in several discussions like this: https://stackoverflow.com/questions/436520/uri-absolutepath-messes-up-path-with-spaces
